### PR TITLE
Rewrite some code for Scala 2.13

### PR DESCRIPTION
**Pull request by dwijnand**
Rewrote some code to avoid some deprecations coming in 2.13.

### CODE REVIEW

**Comment by patriknw** [comment]
any overlap of this and what is in progress in https://github.com/akka/akka/pull/26817 ?

**Comment by dwijnand** [comment]
No, there doesn't seem to be any overlap.

File: akka-testkit/src/test/scala/akka/testkit/TestActorRefSpec.scala [hunk 1]
```diff
@@ -214,7 +214,7 @@ class TestActorRefSpec extends AkkaSpec("disp1.type=Dispatcher") with BeforeAndA
       val a = TestActorRef[WorkerActor]
       val f = a ? "work"
       // CallingThreadDispatcher means that there is no delay
-      f should be('completed)
+      f should be(Symbol("completed"))
```
Comment (patriknw) [inline akka-testkit/src/test/scala/akka/testkit/TestActorRefSpec.scala]: maybe we should change these tests to not use symbols at all?
```
f.isCompleted should be(true)
```

File: akka-actor-tests/src/test/scala/akka/dispatch/StablePriorityDispatcherSpec.scala [hunk 2]
```diff
@@ -26,14 +26,14 @@ object StablePriorityDispatcherSpec {
...
-        case 'Result            => Int.MaxValue
+        case Symbol("Result")   => Int.MaxValue
       }: Any => Int))
 
   class Bounded(@unused settings: ActorSystem.Settings, @unused config: Config)
       extends BoundedStablePriorityMailbox(PriorityGenerator({
         case i: Int if i <= 100 => i // Small integers have high priority
         case _: Int             => 101 // Don't care for other integers
-        case 'Result            => Int.MaxValue
+        case Symbol("Result")   => Int.MaxValue
```
Comment (patriknw) [inline akka-actor-tests/src/test/scala/akka/dispatch/StablePriorityDispatcherSpec.scala]: I would probably have used plain strings in those tests `"Result"`

**Review by patriknw — APPROVED** [review]
LGTM, thanks

**Review by raboof — APPROVED** [review]

**Pull request merged**